### ISSUE TITLE
`ftype` is not assigned when user has multiple files in `datadir`

### ISSUE THREAD

**Issue by nilshg** [floswald/psidR#5]
I had some problems building a panel as I was getting an error message saying `ftype not found`. I think I traced it down to [this line](https://github.com/floswald/psidR/blob/c3c46ab90e74bb3c5e38b76f9314d9070129f1c4/R/build.panel.r#L244) in `build.panel`, which sets `ftype` depending on the data type found in `datadir`. The problem was that I had multiple files in my PSID folder, amongst them the .do files used to create the .dta files. As these appear before the .dta files in alphabetical order, `l[1]` was false for all available options, and hence `ftype` was not set.

I'm not entirely sure what the best workaround is - maybe look through the elements in `l` until a match with either of the data formats is found? Alternatively of course you could just add a word of caution to the docs saying that it's best to only have the actual data files in `mydir`.

**Comment by floswald** [comment]
Not sure what's best here. I can't control what people have in their
folders. Could add an argument saying which file type you supply, or leave
and add to docs.

**Comment by nilshg** [comment]
Maybe the solution could be a combination of 

1. Checking all files in the folder for their extension, to make sure the operation doesn't fail simply because of some extra files in the folder (I'd assume that most people have the .do file in the same folder, as it is inside the zipped folder downloaded from UMich) and 
2. raising a more meaningful error message at this point of the code when all checks for file types fail, something like 
````
l <- list.files(datadir)
if (length(l)==0) stop('there is something wrong with the data directory. please check path')
check = 0
ifelse ( (tail(strsplit(l[1],"\\.")[[1]],1) == "dta"),  ftype   <- "stata", (check = check + 1) )
ifelse ( (tail(strsplit(l[1],"\\.")[[1]],1) == "rda"),  ftype   <- "Rdata", (check = check + 1) )
ifelse ( (tail(strsplit(l[1],"\\.")[[1]],1) == "RData"), ftype <- "Rdata", (check = check + 1) )
ifelse ( (tail(strsplit(l[1],"\\.")[[1]],1) == "csv"), ftype   <- "csv", (check = check + 1) )
if (check>3) stop('No dta, rda, RData or csv files found in supplied datadir')
````

**Comment by floswald** [comment]
i see. probably better to wrap that in a `tryCatch` or just `try`. do you feel like submitting a pull request for this?
thanks for pointing out the issue.

**Comment by nilshg** [comment]
Not a heavy R user, so I don't know whether my proposed change is good style, but it works. I opted for assigning to `ftype`the first match with any of the relevant file types found in the directory, and then raising an error if none of the files in the directory are of either of the four.

**Comment by floswald** [comment]
fixed by 4cad351483ae94035a0ff8e9b215d86e92a99aca
